### ISSUE TITLE
Question: negative ground speed

### ISSUE THREAD

**Issue by kubotron** [fhorinek/SkyDrop#250]
Hi,
I am wondering if the ground speed widget will display negative groundspeed if I would be getting pushed back by the wind?
Cheers, Kubo

**Comment by fhorinek** [comment]
It will be possible after we implement magnetic compass

**Comment by kubotron** [comment]
Ok I'm starting work on this one, after some, umm, interesting events involving groundspeed.

**Comment by bubeck** [comment]
I am not sure, what the best display would be, if gps direction is not "forward" (in fly direction). For the following examples, I assume a speed of the glider of 30 km/h and wind with 45 km/h.

 1. If wind comes from the back, this means, you fly 75 km/h in ground speed. This is correctly displayed already.
 2. If wind comes from front, this means, you fly -15 km/h in ground speed. This is somehow easy to implement (and you suggest this feature)
 3. What should be shown, if wind comes from the right (90 degree). The ground speed would be sqrt(30x30+45x45)=54 km/h but somehow to the front left. Should we show +54 km/h or -54 km/h or even 30km/h (because we only take that part of the wind coming from "front")?

So I propose to always show ground speed without +/- (as it currently is). In addition we can draw an arrow showing the relative direction of our ground movement in regard to compass.

For the above examples we would show an arrow...

 1. pointing upwards (in flight direction).
 2. pointing downwards (in "back" direction)
 3. pointing to the upper left

So this arrow would show the pilot, if we "skew" to a direction which we are not heading to. Even on landing with some sidewind, this would show this skew.

So pilot must be careful, if arrow is not pointing "up" (in flight direction). This would mean some skew, especially if going back.

This widget could be made flashing, if the arrrow is not +/- 90 degree, meaning, that you move backward.

What do you think about that?

**Comment by kubotron** [comment]
I am not sure what would be the best UI. Naive first idea was flashing groundspeed indication if the direction is more or less opposite to the heading (case 2) and ignoring case 3. 

A warning chime / horn tone in addition to that would be ideal to me personally. There is work underway with airspace warnings in skydrop, if I'm not mistaken? Maybe that will introduce some UI concept of warning that we can build on.

**Comment by bluearcus** [comment]
I have only once been in a situation where groundspeed vs compass direction might have clarified what was going on (trying to tag goal in a very windy situation). The rest of the time, it's always been clear enough simply from intermittently looking at my groundspeed roughly what the wind was doing, and this monitoring of wind by checking groundspeed is a constant process, just like looking in the mirror when driving. If my groundspeed is very high, or very low... then the wind is approaching trimspeed. 

Showing an estimated wind speed and direction field on screen is more useful than a 'reversing' indicator, which is specific to a particular heading and speed which the pilot is flying at. What exactly is the additional value of a reversing indicator over a windspeed and direction field?

Some other queries... How would the compass input for a reversing indicator (or alarm) be calibrated?I personally have a config which couldn't use compass data, because of magnetic closures on harnesses (my harness has lots of strong magnetic closures in the cockpit area unfortunately). 

I would suggest an option (config on/off) to improve the windspeed calculation using compass heading data. And perhaps an option to show high windspeeds as a flashing value in the windspeed and/or direction widgets, rather than attempting to implement a 'reversing' indicator.

**Comment by kubotron** [comment]
You raise valid points bluearcus. I will definitely test with magnets when I have something. 

Some instruments in fact do display negative groundspeed, so I think it's realistic to do. UI can be as simple as a minus sign before the groundspeed and an audio chime.

As far as use cases, yeah I hope nobody will ever get to use it hehe. But in a tight situation I'd like to have all available info readily available to help decision making and to stack the chances in my favor as much as possible. 

Imagine landing in a tight valley, on full bar and the instrument displaying 2-4 kmh. Looking over the shoulder for some options, not monitoring the ground speed on the instruments. Now throw in the mix powerlines, cable cars, train tracks, industrial buildings, road and a river. 

Over here both the valley wind and the catabatic wind can get to insane levels quickly so it's not a matter of some theoretical scenarios unfortunately

**Comment by bluearcus** [comment]
If you are landing in a tight valley on full bar, you made a series of very significant mistakes some time ago. Having an on screen representation or alarm for the fact that you are going backwards right now will not solve that - believing that it will is kidding yourself and liable to get you hurt.

**Comment by kubotron** [comment]
Yeah I can understand that sentiment. I don't get the point though. I'd rather have negative groundspeed info than not have it, mistake or not.
